### ISSUE TITLE
Unused alt labels sorted per region

### ISSUE THREAD

**Issue by michihdeu** [TravelMapping/DataProcessing#271]
http://travelmapping.net/logs/unusedaltlabels.log

The file is huge and not sorted by region. It's hard for a highway data manager to find his relevant entries.

It would be nice to have a file per region like we have for [NMPs](http://travelmapping.net/logs/nmpbyregion/).

**Comment by yakra** [comment]
Or how about the same single file, just sorted by region?

**Comment by yakra** [comment]
Until the file actually comes pre-sorted, on noreaster you can type
`sort unusedaltlabels.log | less`
to sort the file and then pipe it into a basic viewer that lets you scroll around with arrow keys, Page Up, Page Down, Home and End.

**Comment by michihdeu** [comment]
I just want the file being empty........ Like data check......

I'm able to find the relevant lines but I'm not sure that all highway data managers find their lines nor that they are even aware of it....................... it's just.....

**Comment by jteresco** [comment]
To me, worrying about the unused alt labels is a pretty low priority.  There's little cost to the existing at site update time, and I think zero cost in the DB or front end.  The only time they matter is when a waypoint label is added or changed and it conflicts with an existing unused alt label.

If we wanted to clean them all up, an automated process seems to be in order.  One pass and we're done, then can watch for new ones creeping in as highway data and list files change.

**Comment by yakra** [comment]
@jteresco wrote:
> There's little cost to the existing at site update time,

The only cost is when a label in a .list file line doesn't match a point's primary label. Then, we iterate thru AltLabels. So the only cost here is when unused AltLabels come before used ones in a given point's AltLabel list, and we have to interate past & check the unused ones before matching the used ones. Not a big cost.

@jteresco wrote:
> and I think zero cost in the DB or front end

Yep. AltLabels are handled in the siteupdate side, during the `TravelerList` constructor. None of that info ever makes it into the DB.

@jteresco wrote:
> If we wanted to clean them all up, an automated process seems to be in order. One pass and we're done, then can watch for new ones creeping in as highway data and list files change.

For my part, I've been purposefully leaving a few in:
* PE1, AB1, AB16, MB1 and MB100 all have unused AltLabels, which are still in use in their corresponding TCH routes. I've only eliminated AltLabels unused in *both* routes, to ensure there are no diffs between the corresponding TCH / provincial files.
* Leaving notes:
  - NL NL430 Goo/Yah [once upon a time](https://github.com/TravelMapping/HighwayData/commit/871d4fef62cebf4872325431658494683117ee2e#diff-889f58dcb55c9c89ea84db3379868c63L102) marked where I switched from using Google's coords to using Yahoo's back when initially drafting the file, when Google's cartography in NL was pretty inaccurate.
  - MB MB262 76N +104W [marks both](https://www.google.com/maps/@50.1243385,-99.8439771,3a,15y,2.48h,88.7t/data=!3m9!1e1!3m7!1s7lo3CRRVB8xMeUb1xBu_kA!2e0!7i13312!8i6656!9m2!1b1!2i40) the N/S and E/W position in the section line road grid. Online maps don't have much labeled in the area, so I left that in as a reference for adding nearby points to other routes.
  - Unused AltLabels in ab.ab577.wpt record where I grabbed coords from what shapefiles.
  - ISTR another contributor (froggie maybe?) using AltLabels for the purposes of taking notes as well.

@michihdeu wrote:
> I just want the file being empty........ Like data check......

Same here (beside the fact that I'm purposefully not helping :D ) though the realist in me accepts it probably won't happen soon.

@michihdeu wrote:
> I'm able to find the relevant lines but I'm not sure that all highway data managers find their lines nor that they are even aware of it....................... it's just.....

Making it more user-friendly and easier to find lines for a particular region certainly won't hurt.
With only 916 lines to sort, the time taken should not even be noticeable.

ISTR @michihdeu mentioning this once before, and I was like "Eh, it's deterministic; it's good enough for me." But it's not good enough though. :)
